_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -69,121 +71,140 @@ In order to use the `s3` sink, configure AWS Identity and Access Management (IAM
+
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-`path_prefix` | No | String | The S3 key prefix path to use for objects written to S3. Accepts date-time formatting. For example, you can use `%{yyyy}/%{MM}/%{dd}/%{HH}/` to create hourly folders in S3. The prefix path should end with `/`. By default, Data Prepper writes objects to the root of the S3 bucket.
-
+`path_prefix` | No | String | The S3 key prefix path to use for objects written to S3. Accepts date-time formatting and dynamic injection of values using [Data Prepper expressions](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/). For example, you can use `/${/my_partition_key}/%{yyyy}/%{MM}/%{dd}/%{HH}/` to create hourly folders in S3 based on the `my_partition_key` value. The prefix path should end with `/`. By default, Data Prepper writes objects to the S3 bucket root.
 
-## codec
+## `codec`
 
 The `codec` determines how the `s3` source formats data written to each S3 object.
 
-### avro codec
+### `avro` codec
 
-The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document.
+The `avro` codec writes an event as an [Apache Avro](https://avro.apache.org/) document. Because Avro requires a schema, you may either define the schema or have Data Prepper automatically generate it. Defining your own schema is recommended, as this will allow it to be tailored to your particular use case.
 
-Because Avro requires a schema, you may either define the schema yourself, or Data Prepper will automatically generate a schema.
-In general, you should define your own schema because it will most accurately reflect your needs.
+When you provide your own Avro schema, that schema defines the final structure of your data. Any extra values inside any incoming events that are not mapped in the Arvo schema will not be included in the final destination. Data Prepper does not allow the use of the `include_keys` or `exclude_keys` with a custom schema. This is to avoid confusion between a custom Arvo schema and the `include_keys` or `exclude_keys` sink configurations. 
 
-We recommend that you make your Avro fields use a null [union](https://avro.apache.org/docs/current/specification/#unions).
-Without the null union, each field must be present or the data will fail to write to the sink.
-If you can be certain that each each event has a given field, you can make it non-nullable.
+In cases where your data is uniform, you may be able to automatically generate a schema. Auto-generated schemas are based on the first event that the codec receives. The schema will only contain keys from this event, and you must have all keys present in all events in order for the auto-generated schema to produce a working schema. Auto-generated schemas make all fields nullable. Use the `include_keys` and `exclude_keys` sink configurations to control what data is included in the auto-generated schema.
```

Comment:
Is "in order for the automatically generated schema to produce a working schema" redundant? Do we just mean "in order to automatically generate a working schema"?

Reply:
Yes. Revised.